### PR TITLE
Add X.509 guidance

### PR DESCRIPTION
**Pull request by wduraes**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,17 +3,24 @@
...
-[Azure IoT Hub ESPRESSIF ESP-32](examples/Azure_IoT_Hub_ESP32/readme.md)
+- [Azure IoT Hub ESPRESSIF ESP-32](examples/Azure_IoT_Hub_ESP32/readme.md)
 
-[Azure IoT Hub Realtek AmebaD](examples/Azure_IoT_Hub_RealtekAmebaD/readme.md)
+- [Azure IoT Hub Realtek AmebaD](examples/Azure_IoT_Hub_RealtekAmebaD/readme.md)
+
+What is the difference between **IoT Hub** and **IoT Central** samples?
+
+1. IoT Hub samples will get devices connected directly to [Azure IoT Hub](https://docs.microsoft.com/en-us/azure/iot-hub/iot-concepts-and-iot-hub)
+1. IoT Central samples, will leverage DPS ([Device Provisioning Service](https://docs.microsoft.com/en-us/azure/iot-dps/about-iot-dps)) to provision the device and then connect it to [Azure IoT Central](https://docs.microsoft.com/en-us/azure/iot-central/core/overview-iot-central).
```
Comment (danewalton-msft) [inline README.md]:
```suggestion
1. IoT Central samples will leverage DPS ([Device Provisioning Service](https://docs.microsoft.com/en-us/azure/iot-dps/about-iot-dps)) to provision the device and then connect it to [Azure IoT Central](https://docs.microsoft.com/en-us/azure/iot-central/core/overview-iot-central). 
```

File: README.md [hunk 2]
```diff
@@ -3,17 +3,24 @@
...
 
-[Azure IoT Hub Realtek AmebaD](examples/Azure_IoT_Hub_RealtekAmebaD/readme.md)
+- [Azure IoT Hub Realtek AmebaD](examples/Azure_IoT_Hub_RealtekAmebaD/readme.md)
+
+What is the difference between **IoT Hub** and **IoT Central** samples?
+
+1. IoT Hub samples will get devices connected directly to [Azure IoT Hub](https://docs.microsoft.com/en-us/azure/iot-hub/iot-concepts-and-iot-hub)
+1. IoT Central samples, will leverage DPS ([Device Provisioning Service](https://docs.microsoft.com/en-us/azure/iot-dps/about-iot-dps)) to provision the device and then connect it to [Azure IoT Central](https://docs.microsoft.com/en-us/azure/iot-central/core/overview-iot-central). 
+
+Please notice that provision through DPS is mandatory for IoT Central scenarios, but DPS can also be used for IoT Hub devices as well.
```
Comment (danewalton-msft) [inline README.md]:
```suggestion
Please note that provisioning through DPS is mandatory for IoT Central scenarios, but DPS can also be used for IoT Hub devices as well.
```

File: examples/Azure_IoT_Central_ESP32/readme.md [hunk 3]
```diff
@@ -76,11 +76,18 @@ To create a device:
 
     ![IoT Central create a device](media/iotcentraldevice-connection-info.png)
 
+    The information provided above will create a device with **Symmetric Key** authentication. If you prefer, you can also use **X.509 certificates**. In order to do that:
+     - In the `Authentication type` box, select `individual enrollment`
+     - In the `Authentication method` box, select `Certificates (X.509)`
```
Comment (danewalton-msft) [inline examples/Azure_IoT_Central_ESP32/readme.md]:
```suggestion
     - In the `Authentication type` box, select `individual enrollment`.
     - In the `Authentication method` box, select `Certificates (X.509)`.
```

File: examples/Azure_IoT_Central_ESP32/readme.md [hunk 4]
```diff
@@ -76,11 +76,18 @@ To create a device:
...
+     - Click on the blue folder icon for both primary and secondary and select your device Certificate.
+
+        > If needed, instructions on how to create an X.509 cert for tests can be found [here](https://github.com/Azure/azure-sdk-for-c/blob/main/sdk/samples/iot/docs/how_to_iot_hub_samples_linux.md#configure-and-run-the-samples) (Step 1). Please note that you might need to install some of the [prerequisites](https://github.com/Azure/azure-sdk-for-c/blob/main/sdk/samples/iot/docs/how_to_iot_hub_samples_linux.md#prerequisites) like OpenSSL.
```
Comment (danewalton-msft) [inline examples/Azure_IoT_Central_ESP32/readme.md]:
```suggestion
        > If needed, instructions on how to create an X.509 cert for testing can be found [here](https://github.com/Azure/azure-sdk-for-c/blob/main/sdk/samples/iot/docs/how_to_iot_hub_samples_linux.md#configure-and-run-the-samples) (Step 1). Please note that you might need to install some of the [prerequisites](https://github.com/Azure/azure-sdk-for-c/blob/main/sdk/samples/iot/docs/how_to_iot_hub_samples_linux.md#prerequisites) like OpenSSL.
```

File: examples/Azure_IoT_Central_ESP32/readme.md [hunk 5]
```diff
@@ -99,7 +106,17 @@ To create a device:
 
 4. Configure the sample.
 
-    Enter your Azure IoT Central and device information into the sample's `iot_configs.h`.
+    Enter your Azure IoT Hub and device information into the sample's `iot_configs.h`:
+    - Add your Wi-Fi SSID to `IOT_CONFIG_WIFI_SSID`
+    - Add your Wi-Fi password to `IOT_CONFIG_WIFI_PASSWORD`
+    - Add you IoTHub Name to `DPS_ID_SCOPE`
```
Comment (danewalton-msft) [inline examples/Azure_IoT_Central_ESP32/readme.md]:
```suggestion
    - Add you Device Provisioning Scope ID to `DPS_ID_SCOPE`
```

File: examples/Azure_IoT_Central_ESP32_AzureIoTKit/readme.md [hunk 6]
```diff
@@ -104,7 +110,17 @@ To create a device:
 
 4. Configure the sample.
 
-    Enter your Azure IoT Central and device information into the sample's `iot_configs.h`.
+    Enter your Azure IoT Hub and device information into the sample's `iot_configs.h`:
+    - Add your Wi-Fi SSID to `IOT_CONFIG_WIFI_SSID`
+    - Add your Wi-Fi password to `IOT_CONFIG_WIFI_PASSWORD`
+    - Add you IoTHub Name to `DPS_ID_SCOPE`
```
Comment (danewalton-msft) [inline examples/Azure_IoT_Central_ESP32_AzureIoTKit/readme.md]:
```suggestion
    - Add you Device Provisioning Scope ID to `DPS_ID_SCOPE`
```

File: examples/Azure_IoT_Central_ESP32_AzureIoTKit/readme.md [hunk 7]
```diff
@@ -104,7 +110,17 @@ To create a device:
...
+    - Add your Device ID to `IOT_CONFIG_DEVICE_ID`
+    - If using **X.509 Cert**:
+        - Uncomment the `#define IOT_CONFIG_USE_X509_CERT`
+        - Add your cert to `IOT_CONFIG_USE_X509_CERT`
+        - Add your cert PK to `IOT_CONFIG_DEVICE_CERT_PRIVATE_KEY`
```
Comment (danewalton-msft) [inline examples/Azure_IoT_Central_ESP32_AzureIoTKit/readme.md]:
```suggestion
        - Add your cert private key to `IOT_CONFIG_DEVICE_CERT_PRIVATE_KEY`
```

File: examples/Azure_IoT_Central_ESP32/readme.md [hunk 8]
```diff
@@ -99,7 +106,17 @@ To create a device:
...
+    - Add your Device ID to `IOT_CONFIG_DEVICE_ID`
+    - If using **X.509 Cert**:
+        - Uncomment the `#define IOT_CONFIG_USE_X509_CERT`
+        - Add your cert to `IOT_CONFIG_USE_X509_CERT`
+        - Add your cert PK to `IOT_CONFIG_DEVICE_CERT_PRIVATE_KEY`
```
Comment (danewalton-msft) [inline examples/Azure_IoT_Central_ESP32/readme.md]:
```suggestion
        - Add your cert private key to `IOT_CONFIG_DEVICE_CERT_PRIVATE_KEY`
```

**Review by danewalton-msft — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (wduraes) [reply README.md]: thanks for taking the time to add all the suggestions! much appreciated!!

**Pull request merged**